src/github_client.rs

```diff
@@ -8,20 +9,29 @@ use futures::future::join_all;
 /// https://developer.github.com/v3/activity/notifications/
 const GITHUB_API_URL: &str = "https://api.github.com/notifications";
 
-/// A client to get messages from the github notification API
+/// A client to get messages from github—or perhaps one day another service's—notification API
 ///
 /// # Arguments
 ///
-/// * `client` - The reqwests client that gets from the github API
+/// * `client` - The `reqwests` client that gets from the github API
 /// * `username` - A github username associated with the token
 /// * `token` - A github personal access token with notification, repo, and non-admin read scopes
-pub struct GithubClient {
+pub struct NotificationClient {
     client: Client,
     username: String,
     token: String,
 }
 
-impl GithubClient {
+#[async_trait]
+pub trait GithubNotificationClient {
+    /// Gets all notifications since a timestamp and bundles them with a human-usable HTML URL.
+    async fn get_humanized_notifications(
```

Comment:
this name is bad and I should feel bad about it and/or which functions I chose to traitify—this is kind of unsatisfying on its own

Reply:
Better than anything coming to me right now so... Good enough for now!